[boost-process:x64-windows] build failure

**Host Environment**
- OS: Windows
- Compiler: Visual Studio 2019

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install boost-process:x64-windows`

**Failure logs** 
C:\dev\vcpkg\installed\x64-windows\include\boost\process\pipe.hpp(280): warning C4244: “参数”: 从“ptrdiff_t”转换到“int”，可能丢失数据
  C:\dev\vcpkg\installed\x64-windows\include\boost/process/pipe.hpp(261): note: 在编译 类 模板 成员函数“bool boost::process::basic_pipebuf<char,std::char_traits<char>>::_write_impl(void)”时
  C:\dev\vcpkg\installed\x64-windows\include\boost/process/pipe.hpp(189): note: 查看对正在编译的函数 模板 实例化“bool boost::process::basic_pipebuf<char,std::char_traits<char>>::_write_impl(void)”的引用
  C:\dev\vcpkg\installed\x64-windows\include\boost/process/pipe.hpp(298): note: 查看对正在编译的 类 模板 实例化“boost::process::basic_pipebuf<char,std::char_traits<char>>”的引用
  ..\..\..\main.cpp(8): note: 查看对正在编译的 类 模板 实例化“boost::process::basic_ipstream<char,std::char_traits<char>>”的引用
  [2/2] Linking CXX executable main.exe
  FAILED: main.exe 
  cmd.exe /C "cd . && "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E vs_link_exe --intdir=CMakeFiles\main.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~4\2019\COMMUN~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\link.exe /nologo CMakeFiles\main.dir\main.cpp.obj  /out:main.exe /implib:main.lib /pdb:main.pdb /version:0.0  /machine:x64 /debug /INCREMENTAL /subsystem:console  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cmd.exe /C "cd /D C:\Users\huangxianyun\Desktop\source\filesystem\out\build\x64-Debug && powershell -noprofile -executionpolicy Bypass -file C:/dev/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary C:/Users/huangxianyun/Desktop/source/filesystem/out/build/x64-Debug/main.exe -installedDir C:/dev/vcpkg/installed/x64-windows/debug/bin -OutVariable out""
  LINK Pass 1: command "C:\PROGRA~2\MICROS~4\2019\COMMUN~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\link.exe /nologo CMakeFiles\main.dir\main.cpp.obj /out:main.exe /implib:main.lib /pdb:main.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:console kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\main.dir/intermediate.manifest CMakeFiles\main.dir/manifest.res" failed (exit code 1120) with the following output:
C:\Users\huangxianyun\Desktop\source\filesystem\out\build\x64-Debug\main.cpp.obj : error LNK2019: 无法解析的外部符号 "__declspec(dllimport) void __cdecl boost::filesystem::path_traits::convert(char const *,char const *,class std::basic_string<wchar_t,struct std::char_traits<wchar_t>,class std::allocator<wchar_t> > &,class std::codecvt<wchar_t,char,struct _Mbstatet> const &)" (__imp_?convert@path_traits@filesystem@boost@@YAXPEBD0AEAV?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AEBV?$codecvt@_WDU_Mbstatet@@@5@@Z)，函数 "void __cdecl boost::filesystem::path_traits::convert(char const *,char const *,class std::basic_string<wchar_t,struct std::char_traits<wchar_t>,class std::allocator<wchar_t> > &)" (?convert@path_traits@filesystem@boost@@YAXPEBD0AEAV?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@@Z) 中引用了该符号
C:\Users\huangxianyun\Desktop\source\filesystem\out\build\x64-Debug\main.cpp.obj : error LNK2019: 无法解析的外部符号 "__declspec(dllimport) void __cdecl boost::filesystem::path_traits::convert(wchar_t const *,wchar_t const *,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &,class std::codecvt<wchar_t,char,struct _Mbstatet> const &)" (__imp_?convert@path_traits@filesystem@boost@@YAXPEB_W0AEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV?$codecvt@_WDU_Mbstatet@@@5@@Z)，函数 "public: class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const __cdecl boost::filesystem::path::string(class std::codecvt<wchar_t,char,struct _Mbstatet> const &)const " (?string@path@filesystem@boost@@QEBA?BV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV?$codecvt@_WDU_Mbstatet@@@5@@Z) 中引用了该符号
C:\Users\huangxianyun\Desktop\source\filesystem\out\build\x64-Debug\main.cpp.obj : error LNK2019: 无法解析的外部符号 "__declspec(dllimport) public: class boost::filesystem::path & __cdecl boost::filesystem::path::operator/=(class boost::filesystem::path const &)" (__imp_??_0path@filesystem@boost@@QEAAAEAV012@AEBV012@@Z)，函数 "class boost::filesystem::path __cdecl boost::filesystem::operator/(class boost::filesystem::path const &,class boost::filesystem::path const &)" (??Kfilesystem@boost@@YA?AVpath@01@AEBV201@0@Z) 中引用了该符号
C:\Users\huangxianyun\Desktop\source\filesystem\out\build\x64-Debug\main.cpp.obj : error LNK2019: 无法解析的外部符号 "__declspec(dllimport) public: static class std::codecvt<wchar_t,char,struct _Mbstatet> const & __cdecl boost::filesystem::path::codecvt(void)" (__imp_?codecvt@path@filesystem@boost@@SAAEBV?$codecvt@_WDU_Mbstatet@@@std@@XZ)，函数 "void __cdecl boost::filesystem::path_traits::convert(char const *,char const *,class std::basic_string<wchar_t,struct std::char_traits<wchar_t>,class std::allocator<wchar_t> > &)" (?convert@path_traits@filesystem@boost@@YAXPEBD0AEAV?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@@Z) 中引用了该符号
C:\Users\huangxianyun\Desktop\source\filesystem\out\build\x64-Debug\main.exe : fatal error LNK1120: 4 个无法解析的外部命令
  ninja: build stopped: subcommand failed.

**CMakeList.txt**
```cmake
cmake_minimum_required (VERSION 3.8)
project ("filesystem")
add_executable (main "main.cpp" )
find_path(BOOST_PROCESS_INCLUDE_DIRS "boost/process.hpp")
target_include_directories(main PRIVATE ${BOOST_PROCESS_INCLUDE_DIRS})
```

**CPP Source Code**
```cpp
#include <iostream>
#include <string>
#include <boost/process.hpp>

int main()
{
    std::string cmd = "C:\\Windows\\System32\\cmd.exe";
    boost::process::ipstream out;
    boost::process::child c(cmd, boost::process::std_out > out);
    c.wait();
}
```

--- a/ports/boost-modular-build-helper/CONTROL
+++ b/ports/boost-modular-build-helper/CONTROL
@@ -1,3 +1,4 @@
 Source: boost-modular-build-helper
 Version: 1.74.0
+Port-Version: 1
 Build-Depends: boost-uninstall

--- a/ports/boost-modular-build-helper/boost-modular-build.cmake
+++ b/ports/boost-modular-build-helper/boost-modular-build.cmake
@@ -148,6 +148,8 @@ function(boost_modular_build)
         if(NOT EXISTS ${CURRENT_PACKAGES_DIR}/lib)
             message(FATAL_ERROR "No libraries were produced. This indicates a failure while building the boost library.")
         endif()
+
+        file(INSTALL ${CURRENT_INSTALLED_DIR}/share/boost-build/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
         return()
     endif()
 
@@ -446,4 +448,5 @@ function(boost_modular_build)
     endif()
 
     vcpkg_copy_pdbs()
+    file(INSTALL ${CURRENT_INSTALLED_DIR}/share/boost-build/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
 endfunction()

--- a/ports/boost-modular-build-helper/portfile.cmake
+++ b/ports/boost-modular-build-helper/portfile.cmake
@@ -7,6 +7,7 @@ file(
         ${CMAKE_CURRENT_LIST_DIR}/nothing.bat
         ${CMAKE_CURRENT_LIST_DIR}/user-config.jam
         ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt
+        ${CMAKE_CURRENT_LIST_DIR}/usage
     DESTINATION
         ${CURRENT_PACKAGES_DIR}/share/boost-build
 )

--- a/ports/boost-modular-build-helper/usage
+++ b/ports/boost-modular-build-helper/usage
@@ -1,0 +1,5 @@
+The package boost is compatible with built-in CMake targets:
+
+    find_package(Boost REQUIRED [COMPONENTS <libs>...])
+    target_link_libraries(main PRIVATE ${Boost_LIBRARIES})
+    target_include_directories(main PRIVATE ${Boost_INCLUDE_DIRS})

--- a/ports/boost-modular-build-helper/usage
+++ b/ports/boost-modular-build-helper/usage
@@ -1,5 +1,4 @@
 The package boost is compatible with built-in CMake targets:
 
     find_package(Boost REQUIRED [COMPONENTS <libs>...])
-    target_link_libraries(main PRIVATE ${Boost_LIBRARIES})
-    target_include_directories(main PRIVATE ${Boost_INCLUDE_DIRS})
+    target_link_libraries(main PRIVATE Boost::boost Boost::<lib1> Boost::<lib2> ...)

--- a/ports/boost-vcpkg-helpers/CONTROL
+++ b/ports/boost-vcpkg-helpers/CONTROL
@@ -1,5 +1,5 @@
 Source: boost-vcpkg-helpers
 Version: 7
-Port-Version: 1
+Port-Version: 2
 Description: a set of vcpkg-internal scripts used to modularize boost
 Build-Depends: boost-uninstall

--- a/ports/boost-vcpkg-helpers/boost-modular-headers.cmake
+++ b/ports/boost-vcpkg-helpers/boost-modular-headers.cmake
@@ -19,6 +19,6 @@ function(boost_modular_headers)
         FILENAME "boost_LICENSE_1_0.txt"
         SHA512 d6078467835dba8932314c1c1e945569a64b065474d7aced27c9a7acc391d52e9f234138ed9f1aa9cd576f25f12f557e0b733c14891d42c16ecdc4a7bd4d60b8
     )
-
+    file(INSTALL ${CURRENT_INSTALLED_DIR}/share/boost-vcpkg-helpers/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
     file(INSTALL ${ARCHIVE} DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
 endfunction()

--- a/ports/boost-vcpkg-helpers/portfile.cmake
+++ b/ports/boost-vcpkg-helpers/portfile.cmake
@@ -1,7 +1,8 @@
 
-file(
-    COPY ${CMAKE_CURRENT_LIST_DIR}/boost-modular-headers.cmake
-    DESTINATION ${CURRENT_PACKAGES_DIR}/share/boost-vcpkg-helpers
+file(COPY
+    ${CMAKE_CURRENT_LIST_DIR}/boost-modular-headers.cmake
+    ${CMAKE_CURRENT_LIST_DIR}/usage
+    DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT}
 )
 
 set(VCPKG_POLICY_EMPTY_PACKAGE enabled)

--- a/ports/boost-vcpkg-helpers/usage
+++ b/ports/boost-vcpkg-helpers/usage
@@ -1,0 +1,5 @@
+The package boost is compatible with built-in CMake targets:
+
+    find_package(Boost REQUIRED [COMPONENTS <libs>...])
+    target_link_libraries(main PRIVATE ${Boost_LIBRARIES})
+    target_include_directories(main PRIVATE ${Boost_INCLUDE_DIRS})

--- a/ports/boost-vcpkg-helpers/usage
+++ b/ports/boost-vcpkg-helpers/usage
@@ -1,5 +1,4 @@
 The package boost is compatible with built-in CMake targets:
 
     find_package(Boost REQUIRED [COMPONENTS <libs>...])
-    target_link_libraries(main PRIVATE ${Boost_LIBRARIES})
-    target_include_directories(main PRIVATE ${Boost_INCLUDE_DIRS})
+    target_link_libraries(main PRIVATE Boost::boost Boost::<lib1> Boost::<lib2> ...)

